src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1454,6 +1455,32 @@ internal void ClearTransaction()
 
         #region Cancel
 
+        internal void PerformUserCancellation()
+        {
+            _userCancellationRequested = true;
+
+            if (AttemptPostgresCancellation && SupportsPostgresCancellation)
+            {
+                Debug.Assert(Settings.CancellationTimeout >= 0);
```

Comment:
Isn't it possible to request user cancellation, while cancellation timeout is -1?

Reply:
We may have to add some tests for this behavior.

Reply:
User cancellation yes - but PG cancellation no (because the meaning of CancellationTimeout == -1 is to disable PG cancellation...).

But I agree that we're missing tests for this...

Reply:
But we do get here, if CancellationTimeout == -1, right?

Reply:
Also, isn't -1 is to fail immediately (we still send the cancellation request)?

Reply:
You're right - I forgot that we send in any case, even if we close immediately.